libs/SalesforceSDK/src/com/salesforce/androidsdk/util/test/TestCredentials.java

```diff
@@ -51,18 +56,25 @@
     public static String PHOTO_URL;
 
     public static void init(Context ctx) {
-        API_VERSION = ApiVersionStrings.getVersionNumber(ctx);
-        ACCOUNT_TYPE = ctx.getString(R.string.account_type);
-        ORG_ID = ctx.getString(R.string.org_id);
-        USERNAME = ctx.getString(R.string.username);
-        ACCOUNT_NAME = ctx.getString(R.string.account_name);
-        USER_ID = ctx.getString(R.string.user_id);
-        LOGIN_URL = ctx.getString(R.string.login_url);
-        INSTANCE_URL = ctx.getString(R.string.instance_url);
-        COMMUNITY_URL = ctx.getString(R.string.community_url);
-        IDENTITY_URL = ctx.getString(R.string.identity_url);
-        CLIENT_ID = ctx.getString(R.string.remoteAccessConsumerKey);
-        REFRESH_TOKEN = ctx.getString(R.string.oauth_refresh_token);
-        PHOTO_URL = ctx.getString(R.string.photo_url);
+        try {
+            JSONObject json = new JSONObject(ResourceReaderHelper.readAssetFile(ctx, "test_credentials.json"));
+
+            API_VERSION = ApiVersionStrings.getVersionNumber(ctx);
+            ACCOUNT_TYPE = ctx.getString(R.string.account_type);
+            ORG_ID = json.getString("organization_id");
+            USERNAME = json.getString("username");
+            ACCOUNT_NAME = json.getString("display_name");
+            USER_ID = json.getString("user_id");
+            LOGIN_URL = json.getString("test_login_domain");
+            INSTANCE_URL = json.getString("instance_url");
+            COMMUNITY_URL = json.optString("community_url", INSTANCE_URL /* in case the test_credentials.json was obtained for a user / org without community setup */);
```

Comment:
If you generate test_credentials.json by login to an org without community setup, it won't have community_url.
In that case we just use the instance_url.